### PR TITLE
Refactor `api.php`

### PR DESCRIPTION
**Pull request by everly-gif**
## What does this PR do?

Refactors `api.php`

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: app/controllers/shared/api.php [hunk 1]
```diff
@@ -1,30 +1,24 @@
...
-    /** @var Utopia\Registry\Registry $register */
-    /** @var Appwrite\Event\Event $events */
-    /** @var Appwrite\Event\Audit $audits */
-    /** @var Appwrite\Event\Mail $mails */
-    /** @var Appwrite\Stats\Stats $usage */
-    /** @var Appwrite\Event\Event $deletes */
-    /** @var Appwrite\Event\Event $database */
-    /** @var Appwrite\Event\Event $functions */
-    /** @var Utopia\Database\Database $dbForProject */
+App::init(function (App $utopia, Request $request, Response $response, Document $project, Document $user, Event $events, Audit $audits, Mail $mails, Stats $usage, Delete $deletes, Event $database, Database $dbForProject, bool $mode) {
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:21]:
```suggestion
App::init(function (App $utopia, Request $request, Response $response, Document $project, Document $user, Event $events, Audit $audits, Mail $mails, Stats $usage, Delete $deletes, Event $database, Database $dbForProject, string $mode) {
```

Comment (christyjacob4) [reply app/controllers/shared/api.php:21]: Mode is a string as seen here
https://github.com/appwrite/appwrite/blob/master/app/init.php#L71-L72

File: app/controllers/shared/api.php [hunk 2]
```diff
@@ -173,18 +164,7 @@
...
-    /** @var Appwrite\Utopia\Response $response */
-    /** @var Utopia\Database\Document $project */
-    /** @var Appwrite\Event\Event $events */
-    /** @var Appwrite\Event\Audit $audits */
-    /** @var Appwrite\Stats\Stats $usage */
-    /** @var Appwrite\Event\Delete $deletes */
-    /** @var Appwrite\Event\Database $database */
-    /** @var bool $mode */
-    /** @var Utopia\Database\Database $dbForProject */
+App::shutdown(function (App $utopia, Request $request, Response $response, Document $project, Event $events, Audit $audits, Stats $usage, Delete $deletes, Event $database, bool $mode, Database $dbForProject) {
```
Comment (christyjacob4) [inline app/controllers/shared/api.php:167]:
```suggestion
App::shutdown(function (App $utopia, Request $request, Response $response, Document $project, Event $events, Audit $audits, Stats $usage, Delete $deletes, Event $database, string $mode, Database $dbForProject) {
```

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**